tests/sa/test_sa_compiled_cache.py

```diff
@@ -0,0 +1,138 @@
+import asyncio
+from aiomysql import sa
+from sqlalchemy import bindparam
+
+import os
+import unittest
+
+from sqlalchemy import MetaData, Table, Column, Integer, String
+
+meta = MetaData()
+tbl = Table('sa_tbl_cache_test', meta,
+            Column('id', Integer, nullable=False,
+                   primary_key=True),
+            Column('val', String(255)))
+
+
+class TestCompiledCache(unittest.TestCase):
+    def setUp(self):
+        self.loop = asyncio.new_event_loop()
+        asyncio.set_event_loop(None)
+        self.host = os.environ.get('MYSQL_HOST', 'localhost')
+        self.port = int(os.environ.get('MYSQL_PORT', 3306))
+        self.user = os.environ.get('MYSQL_USER', 'root')
+        self.db = os.environ.get('MYSQL_DB', 'test_pymysql')
+        self.password = os.environ.get('MYSQL_PASSWORD', '')
+        self.engine = self.loop.run_until_complete(self.make_engine())
+        self.loop.run_until_complete(self.start())
+
+    def tearDown(self):
+        self.engine.terminate()
+        self.loop.run_until_complete(self.engine.wait_closed())
+        self.loop.close()
+
+    async def make_engine(self, **kwargs):
+            return (await sa.create_engine(db=self.db,
+                                           user=self.user,
+                                           password=self.password,
+                                           host=self.host,
+                                           port=self.port,
+                                           loop=self.loop,
+                                           minsize=10,
+                                           **kwargs))
+
+    async def start(self):
+        async with self.engine.acquire() as conn:
+            tx = await conn.begin()
+            await conn.execute("DROP TABLE IF EXISTS "
+                               "sa_tbl_cache_test")
+            await conn.execute("CREATE TABLE sa_tbl_cache_test"
+                               "(id serial, val varchar(255))")
+            await conn.execute(tbl.insert().values(val='some_val_1'))
+            await conn.execute(tbl.insert().values(val='some_val_2'))
+            await conn.execute(tbl.insert().values(val='some_val_3'))
+            await tx.commit()
+
+    def test_cache(self):
+        async def go():
+            cache = dict()
```

Comment:
Pure dict is not option for production system, since it has unbounded size. Should we also provide basic cache implementation also? something like LRU?

Reply:
I think this is up to implementor. LRU could be sufficient in most cases
BTW, sqlalchemy also does not provide any defaults - its only demand is that cache is some dict-like object

Reply:
Makes sense